src/aero_syscall/src/lib.rs

```diff
@@ -835,3 +837,37 @@ pub fn sys_dup2(fd: usize, new_fd: usize, flags: OpenFlags) -> Result<usize, Aer
     let value = syscall3(prelude::SYS_DUP2, fd, new_fd, flags.bits());
     isize_as_syscall_result(value as _)
 }
+
+pub fn sys_ipc_send(pid: usize, message: &[u8]) -> Result<(), AeroSyscallError> {
+    let value = syscall3(
+        prelude::SYS_IPC_SEND,
+        pid,
+        message.as_ptr() as usize,
+        message.len(),
+    );
+    match isize_as_syscall_result(value as _) {
+        Ok(_) => Ok(()),
+        Err(e) => Err(e),
+    }
+}
+
+pub fn sys_ipc_recv<'a>(
+    pid: &mut usize,
+    message: &'a mut [u8],
+    block: bool,
+) -> Result<&'a mut [u8], AeroSyscallError> {
+    let value = syscall4(
+        prelude::SYS_IPC_RECV,
+        pid as *mut usize as usize,
+        message.as_ptr() as usize,
+        message.len(),
+        match block {
+            true => 1,
+            false => 0,
+        },
+    );
+    match isize_as_syscall_result(value as _) {
```

Comment:
I feel like I'm being picky with all those non-rusty paradigms but I would prefer if it was written as follows: `isize_as_syscall_result(value as _).map(|size| &mut message[0..size])`

Reply:
sure!